client/src/main/java/io/pravega/client/stream/impl/SegmentSelector.java

```diff
@@ -126,14 +130,22 @@ public Segment getSegmentForEvent(String routingKey) {
     }
 
     @Synchronized
-    private List<PendingEvent> updateSegmentsUponSealed(StreamSegments newStreamSegments, Segment sealedSegment,
+    private List<PendingEvent> updateSegmentsUponSealed(StreamSegmentsWithPredecessors successors, Segment sealedSegment,
                                                         Consumer<Segment> segmentSealedCallback) {
-        currentSegments = newStreamSegments;
-        createMissingWriters(segmentSealedCallback, newStreamSegments.getDelegationToken());
-        log.debug("Fetch unacked events for segment: {}, and adding new segments {}", sealedSegment, newStreamSegments);
-        List<PendingEvent> toResend = writers.get(sealedSegment).getUnackedEventsOnSeal();
-        writers.remove(sealedSegment); //remove this sealed segment writer.
-        return toResend;
+        if (successors == null) {
```

Comment:
This really should be split into two methods as there is no common code between the if and else case. 
Given that it is only called in one place, this shouldn't be a problem.